app/code/core/Mage/Admin/Model/Observer.php

```diff
@@ -100,11 +100,19 @@ public function actionPreDispatchAdmin($observer)
                             ->setActionName('deniedJson')
                             ->setDispatched(false);
                     } else {
```

Comment:
also, can you merge the `else { if () {} else {} }` with `else if () {} else {}`?